aws/modules/vpc/main.tf

```diff
@@ -0,0 +1,219 @@
+
+  tags = merge(var.tags, {
+    Name = format("private_avilability_zone-%s", local.pri_avilability_zones[count.index])
+    }
+  )
+}
+
+
+resource "aws_route_table_association" "private_route_table_association" {
+  count          = local.pri_az_count
+  subnet_id      = element(aws_subnet.private_subnet.*.id, count.index)
+  route_table_id = element(aws_route_table.private_route_table.*.id, count.index)
+}
+
+#-------------------------------------------------------------------------------
+# Routing table for public subnets
+resource "aws_route_table" "public_route_table" {
+  vpc_id = aws_vpc.vpc.id
+  tags = merge(
+    var.tags,
+    {
+      "Name" = "public_route_table"
+    }
+  )
+}
+
+
+resource "aws_route" "internet_gateway_route" {
+  route_table_id         = aws_route_table.public_route_table.id
+  destination_cidr_block = "0.0.0.0/0"
+  gateway_id             = aws_internet_gateway.internet_gateway.id
+}
+
+# Routing table for private subnets
+resource "aws_route_table" "private_route_table" {
+  count  = var.multi_az_nat_gateway * local.pri_az_count + var.single_nat_gateway * 1
+  vpc_id = aws_vpc.vpc.id
+  tags = merge(
+    var.tags,
+    {
+      az_name = format("private_route_table_az-%s", local.pri_avilability_zones[count.index])
+    }
+  )
+}
+
+
+resource "aws_route" "private_nat_gateway_route" {
+  count                  = var.multi_az_nat_gateway * local.pri_az_count + var.single_nat_gateway * 1
+  route_table_id         = element(aws_route_table.private_route_table.*.id, count.index)
+  nat_gateway_id         = element(aws_nat_gateway.nat_gateway.*.id, count.index)
+  destination_cidr_block = "0.0.0.0/0"
+  depends_on = [
+    aws_route_table.private_route_table,
+    aws_nat_gateway.nat_gateway,
+  ]
+}
+
+#--------------------------------------------------------------------
+
+resource "aws_nat_gateway" "nat_gateway" {
+  count         = var.multi_az_nat_gateway * local.pri_az_count + var.single_nat_gateway * 1
+  subnet_id     = element(aws_subnet.public_subnet.*.id, count.index)
+  allocation_id = element(aws_eip.mod_nat_eip.*.id, count.index)
+  tags          = var.tags
+  depends_on = [
+    aws_internet_gateway.internet_gateway,
+    aws_eip.mod_nat_eip,
+    aws_subnet.public_subnet,
+  ]
+  lifecycle {
+    ignore_changes = [tags]
+  }
+}
+
+
+#-----------------------------------------------------------------------
+
+resource "aws_internet_gateway" "internet_gateway" {
+  vpc_id = aws_vpc.vpc.id
+  tags = merge(
+    var.tags,
+    {
+      "name" = format("%s", aws_vpc.vpc.id)
+    }
+  )
+}
+
+#----------------------------------------------------------------------
+resource "aws_eip" "mod_nat_eip" {
+  count = var.multi_az_nat_gateway * local.pri_az_count + var.single_nat_gateway * 1
+  tags  = var.tags
+  vpc   = true
+}
+
+#-------------------------------------------------------------------------
+
+
+data "http" "workstation-external-ip" {
+  url = "http://ipv4.icanhazip.com"
```

Comment:
What is this for?

Reply:
I have used this data block for internal testing. this data block won't be here.